precompile_process_test.go

```diff
@@ -59,6 +59,35 @@ func testPrecompileProcess(t *testing.T, context spec.G, it spec.S) {
 			Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy"))
 		})
 
+	        context("when a user sets their own RAILS_ENV", func() {
+			it.Before(func() {
+				Expect(os.Setenv("RAILS_ENV", "staging")).To(Succeed())
+			})
+			it.After(func() {
+				Expect(os.Unsetenv("RAILS_ENV")).To(Succeed())
+			})
+			it("runs the bundle exec assets:precompile process while respecting RAILS_ENV", func() {
+				err := precompileProcess.Execute(workingDir)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(executions).To(HaveLen(1))
+				Expect(executions[0].Args).To(Equal([]string{"exec", "rails", "assets:precompile", "assets:clean"}))
+				Expect(executions[0].Env).To(ContainElement("RAILS_ENV=staging"))
+				Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy"))
+			})
+
+			it("runs the bundle exec assets:precompile process while respecting SECRET_KEY_BASE", func() {
+				os.Setenv("SECRET_KEY_BASE", "dummy2")
+				err := precompileProcess.Execute(workingDir)
+				Expect(err).NotTo(HaveOccurred())
+	
+				Expect(executions).To(HaveLen(1))
+				Expect(executions[0].Args).To(Equal([]string{"exec", "rails", "assets:precompile", "assets:clean"}))
+				Expect(executions[0].Env).To(ContainElement("RAILS_ENV=production"))
+				Expect(executions[0].Env).To(ContainElement("SECRET_KEY_BASE=dummy2"))
+			})
```

Comment:
this one should be in its own context outside of the `RAILS_ENV` one, with the same before/after setup for `SECRET_KEY_BASE`